### DISCUSSION TITLE
v2.6.0

### DISCUSSION THREAD

**Discussion by mienaiyami** [mienaiyami/yomikiru#59] [General]
## Downloads

[![Download setup](https://img.shields.io/badge/Windows%20Setup%20(exe)-Manga.Reader-2.6.0-Setup.exe-brightgreen?logo=windows&logoColor=blue)](https://github.com/mienaiyami/react-ts-offline-manga-reader/releases/download/v2.6.0/Manga.Reader-2.6.0-Setup.exe)

[![Download Portable](https://img.shields.io/badge/Windows%20Portable%20(zip)-Manga.Reader-win32-2.6.0-Portable.zip-brightgreen?logo=windows&logoColor=blue)](https://github.com/mienaiyami/react-ts-offline-manga-reader/releases/download/v2.6.0/Manga.Reader-win32-2.6.0-Portable.zip)

---

## Added

- Open app automatically after updating (portable version).
- New shortcuts to switch between `PagePerRow` and `ReaderType`.
- Set max width of images (works only when `Size Clamp` is disabled).
- Search bar:
  - Paste link in search bar to open.
  - Type `..\` to go up directory.
  - Type name of item in results ending with `\` to open it in reader. e.g. `One Piece\`.
  - Type drive name ending with `:\` to change drive. e.g. `D:\`.
- Save toggled state of bookmark / history tab.

## Changed

- Better searching system.
- Prevent some shortcuts if key pressed is held/repeated.
- Disable chapter transition screen in zen mode as well.
- Images in reader are now un-draggable.
- Scroll to current page number after changing `PagePerRow` and `ReaderType`.
- Removed checking for `.bin` file.
- Prevent some shortcuts from working if pressed in combination with `ctrl` key.
- UI enhancements.

## Fixed

- App appear frozen when opening `.zip` or `.cbz` file.

---

#### Todo

- Add canvas based image system for performance.
- Write testing.
- Adjust app for debian.

---

**Full Changelog**: https://github.com/mienaiyami/react-ts-offline-manga-reader/compare/v2.5.2...v2.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mienaiyami/react-ts-offline-manga-reader/releases/tag/v2.6.0'>v2.6.0</a>.</em>

**Comment by omodia** [comment]
canvas based image system for performance?

**Reply by mienaiyami** [reply]
Currently I'm using actual image element show images. Canvas is like windows paint; you can draw on it and add images. I'm planning to use this element to show images. Might even remove stuttering but can't say for sure. 
It will give me full control over images, so i might even be able to control image quality to make reader smoother.

**Reply by omodia** [reply]
Thanks for your hard work. Looking forward to it.

**Reply by omodia** [reply]
> It will give me full control over images, so i might even be able to control image quality to make reader smoother.

But please do not reduce the image quality to make reader smoother.

**Reply by mienaiyami** [reply]
Tt will be user controlled of course.

**Comment by omodia** [comment]
The new UI feels a bit cluttered by the way.

**Reply by mienaiyami** [reply]
Can you them point out so i can make better. 
I also plan to make new system for themes later. I didnt had plans for custom theme when i made this app so themes properties are messed up.
I know myself that im not good at naming thing. You can guess it by this and the fact that i dont have a name for this app yet.
![image](https://user-images.githubusercontent.com/84740082/218502025-3ab2c813-8a78-4758-abb2-c382d36dd35f.png)

**Reply by omodia** [reply]
> You can guess it by this and the fact that i dont have a name for this app yet.

### Some name suggestions after your profile name, 
1. Yami Reader
2. Yami Comic Reader

**Comment by omodia** [comment]
A general question - Have you tried [ChatGPT](https://chat.openai.com)?

**Reply by mienaiyami** [reply]
No, it requires phone number verification.
